Add dietemp stm32l1 config

KOTAROshiota opened the pull request:
Adding die-temp0 aliases for stm32l1.dtsi.
The die_temp_polling sample uses this alias
This PR verified with NUCLEO-L152RE

erwango commented on dts/arm/st/l1/stm32l1.dtsi:
```diff
@@ -17,6 +17,10 @@
 #include <freq.h>
 
 / {
+	aliases {
+		die-temp0 = &die_temp;
+	};
```
According to `samples/sensor/die_temp_polling/sample.yaml`, sample will be built/tested on all boards that have `die-temp0` alias. With this change, this will be all L1 based boards.
Problem is that not all L1 based boards are configured to be able to run this test, and will likely fail or break.

This addition should be done per board, not at soc level.

KOTAROshiota replied:
I got it. I move all changes into boards/arm/nucleo_l152re/nucleo_l152re.dts.

erwango commented on samples/sensor/die_temp_polling/boards/nucleo_l152re.overlay:
```diff
@@ -0,0 +1,15 @@
+ * Copyright (c) 2023 Kotaro SHIOTA
+ */
+
+&die_temp {
+	status = "okay";
+};
+
+&adc1 {
+	pinctrl-0 = <&adc_in0_pa0>;
+	pinctrl-names = "default";
+	status = "okay";
+};
```
Let's move this directly in board.dts.

KOTAROshiota replied:
ditto.

erwango reviewed (changes requested):
Thanks for this change.

Comment on the change location

KOTAROshiota committed:
boards: arm: nucleo_l152re: Add die-temp0 alias

Add die-temp0 alias to die_temp node.
The die_temp_polling sample uses this alias.

Signed-off-by: Kotaro SHIOTA <shiota.kotaro@fujitsu.com>
